<?xml version="1.0" ?>

<launch>
  
  <arg name="robot_ip" default="130.230.37.115" />
  <arg name="load_gripper" default="true" />
  <arg name="launch_rviz" default="true" />
  
  <!-- If needed, broadcast static tf for robot root -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="virtual_joint_broadcaster_1" args="0 0 0 0 0 0 world panda_link0" />



  <include file="$(find franka_control)/launch/franka_control.launch" >
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <arg name="load_gripper" value="$(arg load_gripper)" />
  </include>



  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="position_joint_trajectory_controller"/>

  <include file="$(find franka_example_controllers)/launch/move_group.launch">
    <arg name="load_gripper" value="$(arg load_gripper)" />
  </include>

  <!-- <include file="$(find panda_moveit_config)/launch/moveit_rviz.launch" if="$(arg launch_rviz)" /> -->

  <node name="panda_gripper_control" type="gripper.py" pkg="cobot_controllers" output="screen" />

  <node name="moveit_arm_control" type="arm_controller.py" pkg="cobot_controllers" output="screen" />
  


</launch>
